### PR TITLE
NIFI-12308: Create Python Environment in background thread instead of…

### PR DESCRIPTION
**Pull request by markap14**
… during Processor creation

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-framework-api/src/main/java/org/apache/nifi/components/AsyncLoadedProcessor.java [hunk 1]
```diff
@@ -17,14 +17,18 @@
...
+
+public interface AsyncLoadedProcessor extends Processor {
     default boolean isLoaded() {
         return getState() == LoadState.FINISHED_LOADING;
     }
 
     LoadState getState();
 
     enum LoadState {
+        INITIALIZING_PYTHON_ENVIRONMENT,
```
Comment (exceptionfactory) [inline nifi-framework-api/src/main/java/org/apache/nifi/components/AsyncLoadedProcessor.java:30]: Should this be shortened to `INITIALIZING_ENVIRONMENT`?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for improving the loading process for Python components @markap14, the approach looks good. There is one Checkstyle warning related to an unused import, and I had one question on state naming, but otherwise this ready to go.

File: nifi-framework-api/src/main/java/org/apache/nifi/components/AsyncLoadedProcessor.java (hunk 1 above)
Comment (exceptionfactory) [reply nifi-framework-api/src/main/java/org/apache/nifi/components/AsyncLoadedProcessor.java:30]: Or perhaps `INITIALIZING_PROCESSOR_ENVIRONMENT`? The other states do not mention Python specifically.

**Comment by markap14** [comment]
Thanks @exceptionfactory, pushed a new commit

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the naming adjustment @markap14, the latest version looks good! +1 merging

**Pull request closed without merging**